db/migrate/20150914100056_create_default_project_roles.rb

```diff
@@ -0,0 +1,19 @@
+class CreateDefaultProjectRoles < ActiveRecord::Migration
+  def change
+
+    #Creating default project role relations for non Viewers
+    User.where.not(role_id: Role::VIEWER.id).each { |user|
+      Project.find_each { |project|
+
+        if !UserProjectRole.exists?(project: project, user: user)
+          if user.role_id == Role::ADMIN.id
+            UserProjectRole.create!(project: project, user: user, role_id: ProjectRole::ADMIN.id)
+          elsif user.role_id == Role::DEPLOYER.id
+            UserProjectRole.create!(project: project, user: user, role_id: ProjectRole::DEPLOYER.id)
+          end
+        end
+      }
+    }
```

Comment:
So the idea behind this migration is that we're taking every user that is currently a Deployer or an Admin, and adding an entry in the user_project_roles table for every project? What's the advantage of that?

If we want to downgrade some users, we could selectively give them UserProjectRole entries for the couple of projects they need permissions for, and then change their "baseline" role down to viewer.  But I think that would be more of a manual process, that would require looking at which projects individuals need access to.  This script would add tens of thousands of rows to the table for the instance we're running at Zendesk.